llama_stack/providers/utils/memory/vector_store.py

```diff
@@ -235,6 +246,8 @@ async def insert_chunks(
     ) -> None:
         chunks_to_embed = []
         for i, c in enumerate(chunks):
+            # this should be done in `make_overlapped_chunks` but we do it here for convenience
+            c.metadata["chunk_embedding_model"] = self.vector_db.embedding_model
```

Comment:
This should be revised to only do this for chunks which we are embedding. Precomputed embedding should pass this in.